ibis/backends/singlestoredb/__init__.py

```diff
@@ -0,0 +1,1089 @@
+        return self._client
+
+    def _get_autocommit(self, con) -> bool:
+        """Get autocommit state for both MySQL and HTTP connections.
+
+        Parameters
+        ----------
+        con
+            Connection object (MySQL or HTTP protocol)
+
+        Returns
+        -------
+        bool
+            Current autocommit state
+        """
+        if hasattr(con, "get_autocommit"):
+            # MySQL protocol
+            return con.get_autocommit()
+        elif hasattr(con, "_autocommit"):
+            # HTTP protocol
+            return con._autocommit
+        else:
+            # Default to True if we can't determine
+            return True
+
+    @util.experimental
+    @classmethod
+    def from_connection(cls, con: Connection, /) -> Backend:
+        """Create an Ibis client from an existing connection to a MySQL database.
+
+        Parameters
+        ----------
+        con
+            An existing connection to a MySQL database.
+        """
+        new_backend = cls()
+        new_backend._can_reconnect = False
+        new_backend._client = con
+        new_backend._post_connect()
+        return new_backend
+
+    def _post_connect(self) -> None:
+        with self.con.cursor() as cur:
+            try:
+                cur.execute("SET @@session.time_zone = 'UTC'")
+            except Exception as e:
+                warnings.warn(f"Unable to set session timezone to UTC: {e}")
+
+    @property
+    def show(self) -> Any:
+        """Access to SHOW commands on the server."""
+        return self.con.show
+
+    @property
+    def globals(self) -> Any:
+        """Accessor for global variables in the server."""
+        return self.con.globals
+
+    @property
+    def locals(self) -> Any:
+        """Accessor for local variables in the server."""
+        return self.con.locals
+
+    @property
+    def cluster_globals(self) -> Any:
+        """Accessor for cluster global variables in the server."""
+        return self.con.cluster_globals
+
+    @property
+    def cluster_locals(self) -> Any:
+        """Accessor for cluster local variables in the server."""
+        return self.con.cluster_locals
+
+    @property
+    def vars(self) -> Any:
+        """Accessor for variables in the server."""
+        return self.con.vars
+
+    @property
+    def cluster_vars(self) -> Any:
+        """Accessor for cluster variables in the server."""
+        return self.con.cluster_vars
+
+    @property
+    def current_database(self) -> str:
+        """Return the current database name."""
+        with self.begin() as cur:
+            cur.execute("SELECT DATABASE()")
+            (database,) = cur.fetchone()
+        return database
+
+    @property
+    def database(self) -> str:
+        """Return the current database name (alias for current_database)."""
+        return self.current_database
+
+    @property
+    def dialect(self) -> str:
+        """Return the SQLGlot dialect name."""
+        return "singlestore"
```

Comment:
This should already be defined by the `SQLBackend` parent class, derived from the value of `self.compiler.dialect`